src/components/MultiSelect/MultiSelect.test.tsx

```diff
@@ -0,0 +1,762 @@
+        label="MultiSelect Label"
+        variant="listbox"
+        items={items}
+        isDefaultOpen={true}
+        selectedItems={selectedTestItems}
+        onChange={() => null}
+        onClear={() => null}
+      />
+    );
+
+    const ul = screen.getByRole("listbox");
+    const list = ul[`${Object.keys(ul)[1]}`].children;
+    expect(list).toHaveLength(6);
+  });
+
+  it("should allow user to toggle menu by clicking menu button", () => {
+    const { container } = render(
+      <MultiSelect
+        id="multiselect-listbox-test-id"
+        label="MultiSelect Label"
+        variant="listbox"
+        items={items}
+        selectedItems={selectedTestItems}
+        onChange={() => null}
+        onClear={() => null}
+      />
+    );
+    // When first rendered menu is not expanded
+    expect(screen.queryByRole("listbox")).toBeNull();
+
+    userEvent.click(
+      screen.getByRole("button", {
+        name: /multiselect label/i,
+      })
+    );
+
+    expect(
+      container.querySelector('[aria-expanded="true"][aria-haspopup="listbox"]')
+    ).toBeInstanceOf(HTMLElement);
+    expect(screen.getAllByRole("option")).toHaveLength(6);
+
+    userEvent.click(
+      screen.getByRole("button", {
+        name: /multiselect label/i,
+      })
+    );
+    expect(
+      container.querySelector(
+        '[aria-expanded="false"][aria-haspopup="listbox"]'
+      )
+    ).toBeInstanceOf(HTMLElement);
+    expect(screen.queryAllByRole("option")).toHaveLength(0);
+  });
+
+  it("should call onChange when an item gets selected/unselected", () => {
+    const onChangeMock = jest.fn();
+    render(
+      <MultiSelect
+        id="multiselect-listbox-test-id"
+        label="MultiSelect Label"
+        variant="listbox"
+        items={items}
+        selectedItems={selectedTestItems}
+        onChange={onChangeMock}
+        onClear={() => null}
+      />
+    );
+    // Open menu
+    userEvent.click(
+      screen.getByRole("button", {
+        name: /multiselect label/i,
+      })
+    );
+    expect(screen.getByRole("option", { name: /dogs/i })).toBeInTheDocument();
+    // Check item
+    userEvent.click(screen.getByRole("option", { name: /dogs/i }));
+    expect(onChangeMock).toBeCalledTimes(1);
+    // Check more items
+    userEvent.click(screen.getByRole("option", { name: /plants/i }));
+    userEvent.click(screen.getByRole("option", { name: /colors/i }));
+    expect(onChangeMock).toBeCalledTimes(3);
+    // Uncheck item
+    userEvent.click(screen.getByRole("option", { name: /colors/i }));
+    expect(onChangeMock).toBeCalledTimes(4);
+  });
+
+  it("should render a count button with the correct count, should clear the selectedItems on click ", () => {
+    const { container } = render(
+      <MultiSelectTestListboxComponent id="multiselect-listbox-test-id" />
+    );
+    // Check for the selectedItems count button to not be present
+    expect(
+      container.querySelector("span[role='button']")
```

Comment:
`container.querySelector("span[role='button']")` is repeated a lot. Make it a variable you can reference in this test.